### ISSUE TITLE
Consider moving away from gin framework to standard net/http

### ISSUE THREAD

**Issue by eladmica** [DariaSova/TagGen#28]
Shouldn't be too much work. I think we should do it

https://docs.google.com/presentation/d/10GxCmeI_aJwytELg40L-oqgarFF-SQURr66ItDS26k4/edit

**Comment by tomyang729** [comment]
👍

**Comment by DariaSova** [comment]
out of curiosity what's the reason behind this? (i don't know much about gin)

**Comment by tomyang729** [comment]
I don't know Gin much either. From what I read, Gin is a wrapper on top of net/http, so you can write less codes when doing the same thing, but at the same time, it hides a lot of details of what is really going on. For the purpose of learning golang, it's not a good thing. And also, it would be a challenge if we want to use Gin with some other libraries that are implemented in standard net/http.
Jim is kind of against Gin. Check out his slides from last Go Guild

**Comment by eladmica** [comment]
Yeah basically just for learning. Gin is faster but there probably wouldn't be any difference doing our thing. Also Gin is not that widely used, so you probably won't see it often outside of hootsuite
